### DISCUSSION TITLE
GitHub Actions - Error when doing a git push inside an action

### DISCUSSION THREAD

**Discussion by brpaz** [community/community#26968] [Actions]
<p>Hello.</p>
<p>I am trying to do a git push from a GitHub Action step, but I am having the following error:</p>
<pre><code>fatal: could not read Username for 'https://github.com': No such device or address
</code></pre>
<p>I think it´s related to authentication, but shouldn’t Github Actions handle this automatically?</p>

**Comment by wesleytodd** [comment, marked as answer]
<p>Hey, I was able to get this working.  Here is an example:</p>
<p><a href="https://github.com/pkgjs/gh-pages/blob/master/entrypoint.sh" rel="nofollow noopener">https://github.com/pkgjs/gh-pages/blob/master/entrypoint.sh</a></p>
<p>Basically there are two environment vars you can use <code>GITHUB_ACTOR</code> and <code>GITHUB_TOKEN</code>.  Then use the https push url.  To get the token passed in you need to use the secrets as documented here: <a href="https://help.github.com/en/articles/virtual-environments-for-github-actions#github_token-secret" rel="nofollow noopener">https://help.github.com/en/articles/virtual-environments-for-github-actions#github_token-secret</a></p>
<p>Hope this helps.</p>

**Comment by jakar469** [comment]
<p>download gk bisa di unduh</p>
